botocore/paginate.py

```diff
@@ -323,22 +329,37 @@ def _parse_starting_token(self):
             except ValueError:
                 raise ValueError("Bad starting token: %s" %
                                  self._starting_token)
-        for part in parts:
+        for token_index, part in enumerate(parts):
             if part == 'None':
                 next_token.append(None)
+            elif self._is_dict_token(token_index):
+                try:
+                    # Some services (such as dynamodb) non-string pagination
+                    # tokens, which will fail validation if they aren't
+                    # converted to their proper type.
+                    next_token.append(ast.literal_eval(part))
+                except ValueError:
+                    logger.debug(
+                        "Dict pagination token failed to parse as dict.")
+                    next_token.append(part)
```

Comment:
Out of curiosity, in this case, will the raw `part` still be accepted by the service? If not, shall we use hard failure by `raise ValueError("Dict pagination token failed to parse as dict")`?